app/containers/App.js

```diff
@@ -0,0 +1,221 @@
+    // Use this token to send notification to specific device
+    fcmToken = await firebase.messaging().getToken();
+  }
+
+  async requestPermission() {
+    try {
+        await firebase.messaging().requestPermission();
+        // User has authorised
+        this.getToken();
+    } catch (error) {
+        // User has rejected permissions
+        console.log('permission rejected');
+    }
+  }
+
+
+  // As of Android 8.0 (API Level 26), notifications must specify a Notification Channel or they will not appear.
+  async createNotificationChannel() {
+    // Build a channel
+    const channelId = new firebase.notifications.Android.Channel("Default", "Default", firebase.notifications.Android.Importance.High);
+    // Create the channel
+    await firebase.notifications().android.createChannel(channelId);
+  }
+  
+  // Show notification when app is open
+  showForegroundNotification(notification) {
+      let notification_to_be_displayed = new  firebase.notifications.Notification({
+          data: notification.data,
+          sound: 'default',
+          show_in_foreground: true,
+          title: notification.title,
+          body: notification.body
+      });
+
+      if(Platform.OS == "android") {
+        notification_to_be_displayed
+        .android.setPriority(firebase.notifications.Android.Priority.High)
+        .android.setChannelId("Default")
+        .android.setVibrate(1000);
+      }
+      firebase.notifications().displayNotification(notification_to_be_displayed);
+  }
+
+
+  async createNotificationListeners() {
+    /*
+    * Triggered when a particular notification has been received in foreground
+    * */
+    this.notificationListener = firebase.notifications().onNotification((notification) => {
+        const { title, body } = notification;
+        this.showForegroundNotification(notification);
+    });
+  
+    /*
+    * If your app is in background, you can listen for when a notification is clicked / tapped / opened as follows:
+    * */
+    this.notificationOpenedListener = firebase.notifications().onNotificationOpened((notificationOpen) => {
+      if (notificationOpen) {
+        const { title, body } = notificationOpen.notification;
+      } 
+    });
+  
+    /*
+    * If your app is closed, you can check if it was opened by a notification being clicked / tapped / opened as follows:
+    * */
+    const notificationOpen = await firebase.notifications().getInitialNotification();
+    if (notificationOpen) {
+        const { title, body } = notificationOpen.notification;
+    }
+    
+    /*
+    * Triggered for data only payload in foreground
+    * */
+    this.messageListener = firebase.messaging().onMessage((message) => {
+      //process data message
+      console.log(JSON.stringify(message));
+    });
+  }
+
+  componentWillUnmount() {
+    this.notificationListener();
+    this.notificationOpenedListener();
+    this.messageListener();
+  }
+  
+
+  render() {
+      return (
+        <Fragment>
```

Comment:
Breakdown your UI in smaller functions and components. No UI function should have more than one logical part in it.